frame/include/bli_type_defs.h

```diff
@@ -68,8 +70,8 @@ typedef          uint32_t guint_t;
 typedef           int64_t  gint_t;
 typedef          uint64_t guint_t;
 #else
-typedef   signed long int  gint_t;
-typedef unsigned long int guint_t;
+typedef         ptrdiff_t  gint_t;
+typedef            size_t guint_t;
```

Comment:
I don't think the `stdint.h` dependency is a problem, given that BLIS uses `stdint.h` typedefs pervasively (e.g., 2 lines above this one). And I'm not sure that anyone will ever care about MSVC 2008 anyway -- for those who need MSVC 2008 ABI compatibility, there's mingwpy, and mingwpy isn't restricted to the reference kernels. As for segmented architectures, I guess anyone trying to ported BLIS to one of those should specify a `BLIS_INT_TYPE_SIZE` instead of hoping that this fallback code will somehow do something intelligent :-).

But sure, this is fundamentally just a matter of taste and doesn't matter too much. Maybe @fgvanzee will clarify which way he prefers :-).